src/transactions/transaction.js

```diff
@@ -43,9 +44,12 @@ function createTransaction(recipientId, amount, secret, secondSecret, data, time
 		senderPublicKey: keys.publicKey,
 		timestamp: slots.getTimeWithOffset(timeOffset),
 		asset: {},
-		data,
 	};
 
+	if (data && data.length > 0) {
```

Comment:
What about types? Data here might be stringified when sent to the node, which could lead to unexpected behaviour where the node accepts something the user didn't intend.

Even if validation is happening at the node, do we want to block malformed requests client-side for convenience?